File: 2022/DayNN/Puzzle.py
"""
"""

def part1(data):
    score = ?
    return score


def test_part1():
    data = open('test.txt').readlines()
    answer = ?
    result = part1(data)
    if not answer == result:
        raise AssertionError(f"Part1 FAIL: answer is {answer} not {result}")
    else:
        print("Part1 PASS")
    return True


def part2(data):
    score = ?
    return score

def test_part2():
    data = open('test.txt').readlines()
    answer = ?
    result = part2(data)
    if not answer == result:
        raise AssertionError(f"Part2 FAIL: answer is {answer} not {result}")
    else:
        print("Part2 PASS")
    return True


def main():
  if test_part1():
    data = open('input.txt').readlines()
    score = part1(data)
    print(f"Part 1 score is {score}")

  if test_part2():
    data = open('input.txt').readlines()
    score = part2(data)
    print(f"Part 2 score is {score}")
      
  return


if __name__ == "__main__":
  main()

File: 2022/DayNN/Puzzle.jl
#=

=#

using Logging

function part1(data)
    return false
end

function part2(data)
    return false
end

function main()
    with_logger(ConsoleLogger(stderr, Logging.Debug)) do
        @assert part1(readlines(open("test.txt"))) == true
    end

    open("input.txt") do f
        # read till end of file
        data = readlines(f)
        println("Part 1 is $(part1(data))")
    end

    with_logger(ConsoleLogger(stderr, Logging.Debug)) do
        @assert part2(readlines(open("test.txt"))) == true
    end

    open("input.txt") do f
        # read till end of file
        data = readlines(f)
        println("Part 2 is $(part2(data))")
    end

end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end